### PR TITLE
Allow php 8.3

### PR DESCRIPTION
**Pull request by fezfez**

### CODE REVIEW

**Comment by fezfez** [comment]
mutation fail but not due to this PR

**Comment by Ocramius** [comment]
It looks like most of them are around how exceptions look like: more testing addions, perhaps?

We could also remove the `0` by using the `previous: ` named parameter: ugly, but usable.

**Comment by Ocramius** [comment]
Please wait for #411 before continuing, as @fredden is already on it :D

**Comment by Ocramius** [comment]
@fezfez rebasing should solve CI issues here

**Comment by fezfez** [comment]
> @fezfez rebasing should solve CI issues here

done

**Comment by Ocramius** [comment]
Oof, still one mutant left 😰

**Comment by fredden** [comment]
> Oof, still one mutant left 😰

This is the one squashed in #412. I think we should merge in #388 and then get this rebased as several of the changes here are already being done there.

**Comment by Ocramius** [comment]
Done BTW: more rebasing work 😬

**Comment by fezfez** [comment]
> Done BTW: more rebasing work 😬

done

File: phpunit.xml.dist [hunk 1]
```diff
@@ -1,24 +1,12 @@
...
-    colors="true"
-    convertErrorsToExceptions="true"
-    convertNoticesToExceptions="true"
-    convertWarningsToExceptions="true"
-    verbose="true"
-    stopOnFailure="false"
-    processIsolation="false"
-    backupGlobals="false"
->
+<phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="https://schema.phpunit.de/10.3/phpunit.xsd" bootstrap="./vendor/autoload.php" colors="true" stopOnFailure="false" processIsolation="false" backupGlobals="false" cacheDirectory=".phpunit.cache">
```
Comment (fredden) [inline phpunit.xml.dist]: Can we keep the old formatting here? I find it easier to read with the previous layout.

Comment (fezfez) [reply phpunit.xml.dist]: done

File: phpunit.xml.dist [hunk 2]
```diff
@@ -1,24 +1,20 @@
...
-    convertErrorsToExceptions="true"
-    convertNoticesToExceptions="true"
-    convertWarningsToExceptions="true"
-    verbose="true"
-    stopOnFailure="false"
-    processIsolation="false"
-    backupGlobals="false"
->
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:noNamespaceSchemaLocation="https://schema.phpunit.de/10.3/phpunit.xsd"
```
Comment (fredden) [inline phpunit.xml.dist]: Why has this namespace schema location changed?

File: phpunit.xml.dist (hunk 1 above)
Comment (fredden) [reply phpunit.xml.dist]: Thanks; that helps. :)

File: phpunit.xml.dist (hunk 2 above)
Comment (fezfez) [reply phpunit.xml.dist]: this is produced by using the command provided by phpunit 10 (--migrate-schema)

Comment (Ocramius) [reply phpunit.xml.dist]: We should reference the vendor path here, not a WWW URL

**Review by Ocramius — CHANGES_REQUESTED** [review]
Sorry, one more 😬

**Comment by Ocramius** [comment]
Let's also check if CI is green 😁

File: phpunit.xml.dist (hunk 2 above)
Comment (fezfez) [reply phpunit.xml.dist]: done

**Comment by fezfez** [comment]
all seem good now :)

**Review by Ocramius — APPROVED** [review]
Thanks @fezfez!

**Pull request merged**